src/Tools/Source/RunTests/Program.cs

```diff
@@ -219,25 +240,19 @@ async Task DumpProcess(Process targetProcess, string procDumpExeFilePath, string
                 }
             }
 
-            ConsoleUtil.WriteLine("Roslyn Error: test timeout exceeded, dumping remaining processes");
-            var procDumpInfo = GetProcDumpInfo(options);
-            if (procDumpInfo != null)
+            if (options.CollectDumps && GetProcDumpInfo(options) is { } procDumpInfo)
             {
+                ConsoleUtil.WriteLine("Roslyn Error: test timeout exceeded, dumping remaining processes");
+
                 var counter = 0;
                 foreach (var proc in ProcessUtil.GetProcessTree(Process.GetCurrentProcess()).OrderBy(x => x.ProcessName))
                 {
-                    var dumpDir = PrimaryProcessNames.Contains(proc.ProcessName)
-                        ? procDumpInfo.Value.DumpDirectory
-                        : procDumpInfo.Value.SecondaryDumpDirectory;
+                    var dumpDir = procDumpInfo.DumpDirectory;
                     var dumpFilePath = Path.Combine(dumpDir, $"{proc.ProcessName}-{counter}.dmp");
-                    await DumpProcess(proc, procDumpInfo.Value.ProcDumpFilePath, dumpFilePath);
+                    await DumpProcess(proc, procDumpInfo.ProcDumpFilePath, dumpFilePath);
                     counter++;
                 }
             }
-            else
-            {
-                ConsoleUtil.WriteLine("Could not locate procdump");
-            }
```

Comment:
Do we want to delete this? My worry being it'll be a silent failure now, right?

Reply:
This change unifies that we don't collect dumps unless you specifically ask for them with `--collectdumps`. If you specify that it will then warn you about the inability to find procdump at option parse time and the process won't even start. Hence this should be a non-issue. 

I think the only way to make this stronger is to essentially warn that you didn't specify `--collectdumps` which seems odd since it's an option. We could make it default to on though but the sense I got is we didn't want that to be the default for dev machines since it changes machine state.